metadata-ingestion/src/datahub/cli/cli_utils.py

```diff
@@ -686,3 +686,95 @@ def command(ctx: click.Context) -> None:
         ctx.exit(1)
 
     return command
+
+
+def get_session_login_as(
+    username: str, password: str, frontend_url: str
+) -> requests.Session:
+    session = requests.Session()
+    headers = {
+        "Content-Type": "application/json",
+    }
+    system_auth = get_system_auth()
+    if system_auth is not None:
+        session.headers.update({"Authorization": system_auth})
+    else:
+        data = '{"username":"' + username + '", "password":"' + password + '"}'
+        response = session.post(f"{frontend_url}/logIn", headers=headers, data=data)
+        response.raise_for_status()
+    return session
+
+
+def ensure_valid_gms_url_acryl_cloud(url: str) -> str:
+    if "acryl.io" not in url:
+        return url
+    if url.startswith("http://"):
+        url = url.replace("http://", "https://")
+    if url.endswith("acryl.io"):
+        url = f"{url}/gms"
+    return url
+
+
+def ensure_valid_gms_url(url: str) -> str:
```

Comment:
the function name is kinda confusing - maybe `fixup_gms_url` or something? since it's not really doing any validation

also - I think these methods are great and am planning on reusing them on the DataHubGraph constructor and things

Reply:
done function name.

Majority of this file here needs a cleanup. Plus this is probably should be in the graph as a class method but did not want to try that here as might cause circular dependencies and did not want to get into a large refactor to clean this whole file up.